packages/contracts-bedrock/contracts/test/L1Block.t.sol

```diff
@@ -91,5 +105,141 @@ contract L1BlockTest is CommonTest {
+        assertEq(lb.basefee(), type(uint256).max);
+        assertEq(lb.hash(), keccak256(abi.encode(1)));
+        assertEq(lb.sequenceNumber(), type(uint64).max);
+        assertEq(lb.batcherHash(), bytes32(type(uint256).max));
+        assertEq(lb.l1FeeOverhead(), type(uint256).max);
+        assertEq(lb.l1FeeScalar(), type(uint256).max);
+        assertEq(lb.getL1BlockHash(type(uint64).max), keccak256(abi.encode(1)));
+    }
+
+    function test_getL1BlockHash_upperBound_l1BlockHashNumberOutOfBounds_reverts() external {
+        vm.expectRevert(L1BlockHashNumberOutOfBounds.selector);
+        lb.getL1BlockHash(2); // we set the block number to 1 in the
+        // setup so we are expecting a revert here
+    }
+
+    function test_getL1BlockHash_lowerBound_l1BlockHashNumberOutOfBounds_reverts() external {
+        depositor = lb.DEPOSITOR_ACCOUNT();
+        vm.prank(depositor);
+        lb.setL1BlockValues({
+            _number: uint64(BLOCK_HASHES_SIZE + 1),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: NON_ZERO_HASH,
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+
+        vm.expectRevert(L1BlockHashNumberOutOfBounds.selector);
+        lb.getL1BlockHash(1);
+    }
+
+    function test_getL1BlockHash_hashNotSet_l1BlockHashNumberOutOfBounds_reverts() external {
+        depositor = lb.DEPOSITOR_ACCOUNT();
+        vm.prank(depositor);
+        lb.setL1BlockValues({
+            _number: uint64(3),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: bytes32(0),
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+
+        vm.expectRevert(L1BlockHashNumberOutOfBounds.selector);
+        lb.getL1BlockHash(2);
+    }
+
+    function test_getL1BlockHash_previousBlock_succeeds() external {
+        depositor = lb.DEPOSITOR_ACCOUNT();
+        vm.startPrank(depositor);
+        lb.setL1BlockValues({
+            _number: uint64(2),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: keccak256(abi.encode(2)),
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+
+        lb.setL1BlockValues({
+            _number: uint64(3),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: keccak256(abi.encode(3)),
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+        vm.stopPrank();
+
+        assertEq(lb.getL1BlockHash(2), keccak256(abi.encode(2)));
+        assertEq(lb.getL1BlockHash(3), keccak256(abi.encode(3)));
+    }
+
+    function test_setL1BlockValues_nonDepositor_reverts() external {
```

Comment:
this could be changed to a fuzz test where we pass address as argument. Assume that it is not the depositor and expect reverts